nixos/modules/system/activation/bootspec.nix

```diff
@@ -79,9 +75,18 @@ let
 in
 {
   options.boot.bootspec = {
-    enable = lib.mkEnableOption (lib.mdDoc "Enable generation of RFC-0125 bootspec in $system/bootspec, e.g. /run/current-system/bootspec");
+    enable = lib.mkEnableOption (lib.mdDoc "the generation of RFC-0125 bootspec in $system/boot.json, e.g. /run/current-system/boot.json")
+      // { default = true; };
+    enableValidation = lib.mkEnableOption (
+      lib.mdDoc ''the validation of bootspec documents for each build.
```

Comment:
What are the consequences if this is turned of? Maybe we should enable it, if go is enable.

Reply:
I would prefer not to depend every nixos system on go (not even for build time) by default

Reply:
Especially for bootstrap reasons.

Reply:
If the files do not follow the specification, does that mean boot could be broken? If so I think it is rather desirable to have go in the chain.

Reply:
It's highly unlikely because of those tests that we accept anything broken in nixpkgs, but for people relying on out of tree modules doing complicated stuff, this might offer an interesting amount of guarantees. I feel like this is a strike between cost and guarantees that is acceptable for nixpkgs.